core/src/utils.rs

```diff
@@ -0,0 +1,26 @@
+use gc_arena::Gc;
+
+pub use ruffle_macros::HasPrefixField;
+
+/// A trait indicating that `Self` has `Inner` as an initial prefix.
```

Comment:
nit: I'd elaborate on what that means, e.g. that it has to be a field that has the same address as the struct, is placed at the beginning in the memory, etc.